api/src/main/java/org/pmiops/workbench/api/CohortController.java

```diff
@@ -1,29 +1,174 @@
+    dbCohort.setCreationTime(now);
+    dbCohort.setLastModifiedTime(now);
+    dbCohort = cohortDao.save(dbCohort);
+    return ResponseEntity.ok(TO_CLIENT_COHORT.apply(dbCohort));
+  }
+
+  @Override
+  public ResponseEntity<Void> deleteCohort(String workspaceNamespace, String workspaceId,
+      String cohortId) {
+    org.pmiops.workbench.db.model.Cohort dbCohort = getDbCohort(workspaceNamespace, workspaceId,
+        cohortId);
+    cohortDao.delete(dbCohort);
+    return ResponseEntity.ok(null);
+  }
 
   @Override
-  public ResponseEntity<List<Cohort>> getCohortsInWorkspace(String workspaceNamespace,
+  public ResponseEntity<Cohort> getCohort(String workspaceNamespace, String workspaceId,
+      String cohortId) {
+    org.pmiops.workbench.db.model.Cohort dbCohort = getDbCohort(workspaceNamespace, workspaceId,
+        cohortId);
+    return ResponseEntity.ok(TO_CLIENT_COHORT.apply(dbCohort));
+  }
+
+  @Override
+  public ResponseEntity<CohortListResponse> getCohortsInWorkspace(String workspaceNamespace,
       String workspaceId) {
-    Cohort cohort1 = new Cohort();
-    cohort1.setId("id1");
-    cohort1.setName("name1");
-    cohort1.setCriteria("criteria1");
-    cohort1.setType("type1");
-
-    Cohort cohort2 = new Cohort();
-    cohort2.setId("id2");
-    cohort2.setName("name2");
-    cohort2.setCriteria("criteria2");
-    cohort2.setType("type2");
-
-    return ResponseEntity.ok(ImmutableList.of(cohort1, cohort2));
+    Workspace workspace = getDbWorkspace(workspaceNamespace, workspaceId);
+    CohortListResponse response = new CohortListResponse();
+    Set<org.pmiops.workbench.db.model.Cohort> cohorts = workspace.getCohorts();
+    if (cohorts != null) {
+      response.setItems(cohorts.stream().map(TO_CLIENT_COHORT).collect(Collectors.toList()));
+    }
+    return ResponseEntity.ok(response);
+  }
+
+  @Override
+  public ResponseEntity<Cohort> updateCohort(String workspaceNamespace, String workspaceId,
+      String cohortId, Cohort cohort) {
+    org.pmiops.workbench.db.model.Cohort dbCohort = getDbCohort(workspaceNamespace, workspaceId,
+        cohortId);
+    if (cohort.getType() != null) {
+      dbCohort.setType(cohort.getType());
+    }
+    if (cohort.getName() != null) {
+      dbCohort.setName(cohort.getName());
+    }
+    if (cohort.getDescription() != null) {
+      dbCohort.setDescription(cohort.getDescription());
+    }
+    if (cohort.getCriteria() != null) {
+      dbCohort.setCriteria(cohort.getCriteria());
+    }
+    // TODO: add version, check it here
+    dbCohort = cohortDao.save(dbCohort);
+    return ResponseEntity.ok(TO_CLIENT_COHORT.apply(dbCohort));
+  }
+
+  private Workspace getDbWorkspace(String workspaceNamespace, String workspaceId) {
+    String firecloudName = Workspace.toFirecloudName(workspaceNamespace, workspaceId);
+    Workspace workspace = workspaceDao.findByFirecloudName(firecloudName);
+    if (workspace == null) {
+      // Create a workspace if it doesn't already exist.
```

Comment:
For now, promoting this to a docstring would be good, since it's a big part of what the method does.